joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/RubyImportResolverPass.scala

```diff
@@ -0,0 +1,93 @@
+package io.joern.rubysrc2cpg.passes
+
+import better.files.File
+import io.joern.rubysrc2cpg.deprecated.utils.PackageTable
+import io.joern.x2cpg.Defines as XDefines
+import io.shiftleft.semanticcpg.language.importresolver.*
+import io.joern.x2cpg.passes.frontend.XImportResolverPass
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.Call
+import io.shiftleft.semanticcpg.language.*
+
+import java.io.File as JFile
+import java.util.regex.{Matcher, Pattern}
+class RubyImportResolverPass(cpg: Cpg) extends XImportResolverPass(cpg) {
+
+  private val pathPattern = Pattern.compile("[\"']([\\w/.]+)[\"']")
+
+  override protected def optionalResolveImport(
+    fileName: String,
+    importCall: Call,
+    importedEntity: String,
+    importedAs: String,
+    diffGraph: DiffGraphBuilder
+  ): Unit = {
+
+    resolveEntities(importedEntity, importCall, fileName).foreach(x => evaluatedImportToTag(x, importCall, diffGraph))
+  }
+
+  private def resolveEntities(expEntity: String, importCall: Call, fileName: String): Set[EvaluatedImport] = {
+    // TODO: Currently only working on internal dependencies, will be fixed for external dependencies once the dependency linking is done
+    val expResolvedPath =
+      if (expEntity.contains("."))
+        getResolvedPath(expEntity, fileName)
+      else if (cpg.file.name(s".*$expEntity.rb").nonEmpty)
+        getResolvedPath(s"$expEntity.rb", fileName)
+      else
+        expEntity
+
+    // TODO Limited ResolvedMethod exposure for now, will open up after looking at more concrete examples
+    val finalResolved = {
+      val filePattern = s"${Pattern.quote(expResolvedPath)}\\.?.*"
+      val resolvedTypeDecls = cpg.typeDecl
+        .where(_.file.name(filePattern))
+        .fullName
+        .flatMap(fullName =>
+          Seq(
+            ResolvedTypeDecl(fullName),
+            ResolvedMethod(s"$fullName.${XDefines.ConstructorMethodName}", "new", fullName.split("[.]").lastOption)
+          )
+        )
+        .toSet
+
+      val resolvedModules = cpg.namespaceBlock
+        .whereNot(_.nameExact("<global>"))
```

Comment:
From `import io.shiftleft.semanticcpg.language.types.structure.NamespaceTraversal`
```suggestion
        .whereNot(_.nameExact(NamespaceTraversal.globalNamespaceName))
```